common/client.go

```diff
@@ -365,14 +369,43 @@ func (rc *RestClient) execMethod(method string, dest string, data interface{}, r
 	if err != nil {
 		return err
 	}
-
-	if result == nil {
-		return nil
+	var unmarshalBodyErr error
+	if result != nil {
+		unmarshalBodyErr = json.Unmarshal(body, &result)
 	}
-
-	err = json.Unmarshal(body, &result)
-	if err != nil {
-		return errors.New(fmt.Sprintf("Error %s (%T) when parsing %s", err.Error(), err, body))
+	if resp.StatusCode >= 400 {
```

Comment:
Well in a way yes... But in this case a numeric value is more obvious to show what is being done, whereas one would have to think a bit as to why we are doing >= operation with some status